File: 792-BinarySearch/792-BinarySearch.cpp
// Last updated: 5/20/2026, 11:36:35 PM
#include <iostream>
#include <vector>
#include <cmath>

class Solution {
    public:
        int search(std::vector<int>& nums, int target) {
            int lo = 0, mid = 0, hi = nums.size();

            while(lo < hi) {
                mid = floor(lo + (hi - lo) / 2);

                if (nums[mid] == target) {
                    return mid;
                } else if (nums[mid] > target) { 
                    hi = mid;
                    continue;
                } else if (nums[mid] < target) {
                    lo = mid + 1;
                    continue;
                }

            }
            return -1;
        }
};

File: 792-BinarySearch/792-BinarySearch.ts
// Last updated: 5/20/2026, 11:36:38 PM
function search(nums: number[], target: number): number {
    let lo = 0;
    let hi = nums.length;

    do
    {
        const m = Math.floor(lo + (hi - lo) / 2);
        const v = nums[m];

        if(target === nums[m])
        {
            return m;
        }
        else if (target > nums[m])
        {
            lo = m + 1;
        } 
        else 
        {
            hi = m;
        }

    }while(lo < hi);
    return -1;
};